### PR TITLE
I5550 converting confirm btn to use setUIState

### LINKED ISSUE #5550: Convert setState() in ConfirmButton to setUIState()
The [ConfirmButton](https://github.com/mozilla/addons-frontend/blob/54c4a4ddbfaea24c27620bead2254b183ba8bc61/src/ui/components/ConfirmButton/index.js) component can be converted to using [`setUIState()`](https://github.com/mozilla/addons-frontend/issues/5343).

It's only using `setState()` to show a prompt about what the user is about to confirm.

**For QA**: please make sure that confirm buttons on the collection and user profile screens still work as expected.

Part of https://github.com/mozilla/addons-frontend/issues/1250

### PR DESCRIPTION
**Pull request by rebmullin**
fixes #5550 

I wasn't totally clear if I need to do anything to the HOC. I saw this component in 2 places - the collections and the user profile - and it looks like it works with the changes in both places still. 

let me know what you think : )

### CODE REVIEW

File: src/ui/components/ConfirmButton/index.js [hunk 1]
```diff
@@ -118,8 +117,18 @@ export class ConfirmButtonBase extends React.Component<InternalProps, State> {
   }
 }
 
-const ConfirmButton: React.ComponentType<Props> = compose(translate())(
-  ConfirmButtonBase,
-);
+export const extractId = (ownProps: Props) => {
+  return `confirmButton-${ownProps.className}`;
```
Comment (rebmullin) [inline src/ui/components/ConfirmButton/index.js]: I wasn't totally clear about this - will this work?

Comment (rebmullin) [reply src/ui/components/ConfirmButton/index.js]: I guess i need to change this anyway since classname is not required :/

File: src/ui/components/ConfirmButton/index.js [hunk 2]
```diff
@@ -118,8 +117,18 @@ export class ConfirmButtonBase extends React.Component<InternalProps, State> {
   }
 }
 
-const ConfirmButton: React.ComponentType<Props> = compose(translate())(
-  ConfirmButtonBase,
-);
+export const extractId = (ownProps: Props) => {
+  return `confirmButton-${ownProps.className || ''}`;
```
Comment (rebmullin) [inline src/ui/components/ConfirmButton/index.js]: This doesn't feel ideal to me. Should I be passing in an id to the component? Or should I make this function optional? I didn't feel like that was a good idea either tho :/

File: src/ui/components/ConfirmButton/index.js (hunk 1 above)
Comment (willdurand) [reply src/ui/components/ConfirmButton/index.js]: mmh, do you really need an ID? I think you can write:

```
extractId: () => '',
```

in the HOC declaration.

Comment (rebmullin) [reply src/ui/components/ConfirmButton/index.js]: ahh! ok that will work i think. I wasn't clear if I should be passing in *something*..

File: tests/unit/ui/components/TestConfirmButton.js [hunk 3]
```diff
@@ -89,59 +97,60 @@ describe(__filename, () => {
...
-    root.find(Button).simulate('click', createFakeEvent());
-
-    expect(root.find('.ConfirmButton-cancel-button')).toHaveProp(
-      'buttonType',
-      cancelButtonType,
-    );
-    expect(root.find('.ConfirmButton-cancel-button').children()).toHaveText(
-      cancelButtonText,
-    );
-  });
```
Comment (willdurand) [inline tests/unit/ui/components/TestConfirmButton.js]: Why did you remove this test case?

Comment (rebmullin) [reply tests/unit/ui/components/TestConfirmButton.js]: hmm..good question...let me double check : )

Comment (rebmullin) [reply tests/unit/ui/components/TestConfirmButton.js]: I think I thought this was dupe but I see it's slightly different so I will add this back. thanks for catching this!

File: tests/unit/ui/components/TestConfirmButton.js [hunk 4]
```diff
@@ -107,41 +117,59 @@ describe(__filename, () => {
...
+  it('changes UI state when the confirm button is clicked', () => {
+    const { store } = dispatchClientMetadata();
+    const root = render({ store });
+
+    expect(root.instance().props.uiState.showConfirmation).toEqual(false);
+
+    root.find(Button).simulate('click', createFakeEvent());
+    applyUIStateChanges({ root, store });
+
+    expect(root.instance().props.uiState.showConfirmation).toEqual(true);
```
Comment (willdurand) [inline tests/unit/ui/components/TestConfirmButton.js]: Not sure you really this test case, because it should be already tested by the previous test cases.

Comment (rebmullin) [reply tests/unit/ui/components/TestConfirmButton.js]: ok I was wondering but I saw an explicit test for this in the other place it was already set up. But I can def removed if it's overkill here..

Comment (willdurand) [reply tests/unit/ui/components/TestConfirmButton.js]: > vut I saw an explicit test for this in the other place it was already set up

ah, really? I looked too, and found nothing 😅

Comment (rebmullin) [reply tests/unit/ui/components/TestConfirmButton.js]: sorry I should have posted where I saw this (I was being lazy :/), but [here](https://github.com/mozilla/addons-frontend/blob/master/tests/unit/amo/components/TestEditableCollectionAddon.js#L224)

Comment (willdurand) [reply tests/unit/ui/components/TestConfirmButton.js]: mmh interesting. I'd argue that we should not be testing this because we should focus on the behaviors. If the value of this prop changes, then we should observe "something", in this case a text with two buttons should be either displayed or hidden.

But maybe @kumar303 has a different opinion.
